client/src/rootReducer.js

```diff
@@ -0,0 +1,11 @@
+import {forwardReducerTo} from './utils'
+import getInitialState from './state'
+
+const rootReducer = (state = getInitialState(), action) => {
+  const {reducer, type, path, payload} = action
+  if (!reducer) return state
+  if (!path) throw new Error(`You forgot to set path in action ${type}`)
```

Comment:
I don't mind - you can always make it so that without specifying a path it defaults to [] (I'd change it to that), I can see the convenience in being able to specify it if you need it

Reply:
also ,regarding immutability-helper - I'd give this a bash this time https://github.com/mweststrate/immer